pkg/antctl/raw/multicluster/create/access_token.go

```diff
@@ -0,0 +1,171 @@
+	command.Flags().StringVarP(&o.serviceAccount, "service-account", "", "", "Service Account of the access token")
+	command.Flags().StringVarP(&o.roleBinding, "role-binding", "", "", "RoleBinding of the Service Account")
+
+	return command
+}
+
+func accessTokenRunE(cmd *cobra.Command, args []string) error {
+	if err := accessTokenOpts.validateAndComplete(); err != nil {
+		return err
+	}
+	if len(args) != 1 {
+		return fmt.Errorf("exactly one NAME is required, got %d", len(args))
+	}
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+	restconfigTmpl := rest.CopyConfig(kubeconfig)
+	raw.SetupKubeconfig(restconfigTmpl)
+	k8sClient, err := client.New(kubeconfig, client.Options{Scheme: multiclusterscheme.Scheme})
+	if err != nil {
+		return err
+	}
+
+	serviceAccount := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      accessTokenOpts.serviceAccount,
+			Namespace: accessTokenOpts.namespace,
+		},
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "creating ServiceAccount %s \n", accessTokenOpts.serviceAccount)
+	if err := k8sClient.Create(context.TODO(), &serviceAccount); err != nil {
+		if errors.IsAlreadyExists(err) {
+			fmt.Fprintf(cmd.OutOrStderr(), fmt.Sprintf("the ServiceAccount %s is already exists\n", accessTokenOpts.serviceAccount))
+		} else {
+			return err
+		}
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "creating RoleBinding %s \n", accessTokenOpts.roleBinding)
+	roleBinding := rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      accessTokenOpts.roleBinding,
+			Namespace: accessTokenOpts.namespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     "antrea-mc-member-cluster-role",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      accessTokenOpts.serviceAccount,
+				Namespace: accessTokenOpts.namespace,
+			},
+		},
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "creating RoleBinding %s \n", accessTokenOpts.serviceAccount)
+	if err := k8sClient.Create(context.TODO(), &roleBinding); err != nil {
+		if errors.IsAlreadyExists(err) {
+			fmt.Fprintf(cmd.OutOrStderr(), fmt.Sprintf("the RoleBinding %s is already exists\n", accessTokenOpts.roleBinding))
+		} else {
+			rollback(k8sClient, accessTokenOpts.namespace, accessTokenOpts.serviceAccount, "")
+			return err
+		}
+	}
+	secretName := args[0]
+
+	fmt.Fprintf(cmd.OutOrStdout(), "creating Secret %s \n", secretName)
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: accessTokenOpts.namespace,
+			Annotations: map[string]string{
+				"kubernetes.io/service-account.name": accessTokenOpts.serviceAccount,
+			},
+		},
+		Type: "kubernetes.io/service-account-token",
+	}
+
+	if err := k8sClient.Create(context.TODO(), &secret); err != nil {
+		if errors.IsAlreadyExists(err) {
+			fmt.Fprintf(cmd.OutOrStderr(), fmt.Sprintf("the Secret %s is already exists\n", accessTokenOpts.roleBinding))
```

Comment:
ditto

Reply:
The secret name is first argument `args[0]`, you should use `secretName` instead of `accessTokenOpts.roleBinding`

Reply:
should rollback also be called even when it's `IsAlreadyExists` error?

Reply:
done